tests/puppeteer/campaigns/functional/BO/14_advancedParameters/07_database/sqlManager/01_CRUDSqlQuery.js

```diff
@@ -0,0 +1,210 @@
+require('module-alias/register');
+const {expect} = require('chai');
+const helper = require('@utils/helpers');
+const loginCommon = require('@commonTests/loginBO');
+// Importing pages
+const BOBasePage = require('@pages/BO/BObasePage');
+const LoginPage = require('@pages/BO/login');
+const DashboardPage = require('@pages/BO/dashboard');
+const SqlManagerPage = require('@pages/BO/advancedParameters/database/sqlManager');
+const AddSqlQueryPage = require('@pages/BO/advancedParameters/database/sqlManager/add');
+const ViewQueryManagerPage = require('@pages/BO/advancedParameters/database/sqlManager/view');
+// Import pages
+const SQLQueryFaker = require('@data/faker/sqlQuery');
+const {Tables} = require('@data/demo/sqlTables');
+// Test context imports
+const testContext = require('@utils/testContext');
+
+const baseContext = 'functional_BO_advancedParams_database_CRUDSqlQuery';
+
+let browser;
+let page;
+let numberOfSQLQuery = 0;
+const sqlQueryData = new SQLQueryFaker({tableName: 'ps_alias'});
+const editSqlQueryData = new SQLQueryFaker({name: `edit${sqlQueryData.name}`, tableName: 'ps_access'});
+
+// Init objects needed
+const init = async function () {
+  return {
+    boBasePage: new BOBasePage(page),
+    loginPage: new LoginPage(page),
+    dashboardPage: new DashboardPage(page),
+    sqlManagerPage: new SqlManagerPage(page),
+    addSqlQueryPage: new AddSqlQueryPage(page),
+    viewQueryManagerPage: new ViewQueryManagerPage(page),
+  };
+};
+
+describe('CRUD SQL query', async () => {
+  // before and after functions
+  before(async function () {
+    browser = await helper.createBrowser();
+    page = await helper.newTab(browser);
+    await helper.setDownloadBehavior(page);
+    this.pageObjects = await init();
+  });
+  after(async () => {
+    await helper.closeBrowser(browser);
+  });
+
+  // Login into BO
+  loginCommon.loginBO();
+
+  // Go to database page
+  it('should go to \'Advanced Parameters > Database\' page', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'goToDatabasePageToCreateNewSQLQuery', baseContext);
+    await this.pageObjects.boBasePage.goToSubMenu(
+      this.pageObjects.boBasePage.advancedParametersLink,
+      this.pageObjects.boBasePage.databaseLink,
+    );
+    await this.pageObjects.boBasePage.closeSfToolBar();
+    const pageTitle = await this.pageObjects.sqlManagerPage.getPageTitle();
+    await expect(pageTitle).to.contains(this.pageObjects.sqlManagerPage.pageTitle);
+  });
+
+  it('should reset all filters', async function () {
+    await testContext.addContextItem(this, 'testIdentifier', 'firstResetFilter', baseContext);
+    numberOfSQLQuery = await this.pageObjects.sqlManagerPage.resetAndGetNumberOfLines();
+    if (numberOfSQLQuery !== 0) await expect(numberOfSQLQuery).to.be.above(0);
```

Comment:
I would never use inline condition myself, that make its harder to read, don't you think @nesrineabdmouleh ? Just a suggestion, not heavy

Reply:
Thank you :) I will fix it ^^